lib/pbench/server/api/resources/server_configuration.py

```diff
@@ -113,40 +113,74 @@ def _put(self, params: ApiParams, _) -> Response:
         }
         """
 
-        try:
-            key = params.uri["key"]
-            try:
-                value = params.query["value"]
-            except KeyError:
-                try:
-                    value = params.body["value"]
-                except KeyError:
+        # First, we check for the ways to set an individual config setting by
+        # naming the config key on the URI and specifying a value either with
+        # the "value" either as a query parameter or in the JSON request body.
+        #
+        # We'll complain about JSON request body parameters that are "shadowed"
+        # by the URI or query parameter and might represent client confusion.
+        keydup = set()
+        key = params.uri.get("key")
+        if key:
+            # If we have a key in the URL, the we need a "value" for it, which
+            # we can take either from a query parameter or from the JSON
+            # request payload.
+            if params.body and list(params.body.keys()) != ["value"]:
+                keydup.update(params.body.keys())
+            value = params.query.get("value")
+            if value:
+                if params.body:
+                    keydup.update(params.body.keys())
+            else:
+                value = params.body.get("value")
+                if not value:
                     raise APIAbort(
                         HTTPStatus.BAD_REQUEST,
-                        "Incorrect value specification for key {key!r}",
+                        f"No value found for key system configuration key {key!r}",
                     )
+
+            if len(keydup) > 0:
+                raise APIAbort(
+                    HTTPStatus.BAD_REQUEST,
+                    f"Redundant parameters specified in the JSON request body: {sorted(keydup)!r}",
```

Comment:
> That's a bigger deal when we get value from the body, where I want to complain about any other keys rather than just saying "bad caller ... bad".

I support that, but the blob cited here is just the query parameter.

> I just reversed from the "second style" relying on KeyError in favor of pre-testing. I go back and forth on that philosophical point depending on circumstances.

I concur.  And, for the outer conditional ("if key"), I think using an explicit test is appropriate, because there is no "normal" vs. "exceptional" dichotomy -- both cases are "normal" and we're just want to know which one to execute.  (So, the circumstances there suggest the explicit test philosophy.  😁)  For the inner conditional ("if value"), I could go either way, but I find the explicit test easier to read (although, philosophically, the `try`/`except` is probably a better fit).

> I reversed in this case because you expressed confusion about the logic pattern and this seemed more straightforward to read.

And, I appreciate that (and I agree with it)...but, I couldn't resist representing Peter's point of view, since he's stuck there on PTO and can't stick his own oar in.  🤣